draft-wang-ppm-differential-privacy.md

```diff
@@ -239,11 +254,133 @@ The following considerations are out-of-scope for this document:
+`p(S, D2, R)` is negligible. However, this requires adding more noise, which
+has an adverse impact on utility. Most applications will accept a
+non-negligible bound in order to achieve reasonable utility. See
+{{dp-explainer}} for details.
+
+This notion of `EPSILON`-DP is sometimes referred to as "pure-DP". The
+following is a relaxation of pure-DP, called "approximate-DP", from {{DR14}}. A
+randomized algorithm `R` is called "`(EPSILON, DELTA)`-DP" if for all
+neighboring `D1` and `D2` and all possible aggregate results `S`, the following
+inequality holds:
+
+~~~
+p(S, D1, R) <= exp(EPSILON) * p(S, D2, R) + DELTA
+~~~
 
-TODO
+Compared to pure-DP, approximate-DP loses an additive factor of `DELTA` in the
+bound. This is to account for certain DP mechanisms that have some desirable
+properties, but are not pure-DP. See {{mechanisms}} for details.
+
+Other variants of DP are possible; see the literature review in
+{{dp-explainer}} for details.
+
+## Trust Models
+
+When considering whether a given DP policy is sufficient for DAP, it is not
+enough to consider the mechanisms used in isolation. It is also necessary to
+consider the process by which the policy is executed. In particular, our threat
+model for DAP considers an attacker that participates in the upload,
+aggregation, and collection protocols and may use its resources to attack the
+underlying primitives (VDAF {{!VDAF}}, HPKE {{!RFC9180}}, and the transport
+security protocol).
+
+To account for such attackers, our goal for DAP is "computational-DP" as
+described by {{MPRV09}}, which formalizes the amount of information an active
+attacker can glean about the measurements over the course of its attack on the
+protocol. We consider an attacker that controls the network and statically
+corrupts a subset of the parties.
+
+We say the protocol is pure-DP (resp. approximate-DP) if the view of any
+computationally bounded attacker can be efficiently simulated by a simulator
+that itself is pure-DP (or approximate-DP) as defined above. (The simulator
+takes the measurements as input and outputs a value that is computationally
+indistinguishable from the transcript of the protocol's execution in the
+presence of the attacker.)
+
+Whether this property holds for a given DP policy depends on which parties can
+be trusted to execute the protocol correctly (i.e., which parties were not
+corrupted by the attacker). The DP policies described in this document are
+secure in one of two trust models described below.
+
+### OAMC: One-Aggregator-most-Clients
+
+Assume that most Clients and one Aggregator is honest. The other
+Aggregator and the Collector are controlled by the atacker. The degree of
+privacy provided by the DP policy (i.e., `EPSILON`) degrades as the number of
+honest Clients decreases.
+
+### OA: One-Aggregator
+
+Assume that at least one Aggregator is honest. The other Aggregator(s), the
+Collector, and most Clients are controlled by the attacker.
+
```

Comment:
Should it be "all Clients" instead of "most Clients", since we only trust one Aggregator and nobody else?

Reply:
@wangshan pointed out that we're considering at least one honest Client, since otherwise privacy is completely meaningless :D

Reply:
I see, although I'd say that in that case privacy is trivially true rather than meaningless ("for all neighboring databases in \emptyset, ..."). 

But if we want to consider at least one honest Client, then we could make it explicit in the trust model, and say "all but one Client" instead of "most Clients". The separation between OneAggregator-MostClients and OneAggregator-OneClient seems cleaner than between OneAggregator-MostClients and OneAggregator-FewClients.

Reply:
Well, ideally what we want is that that DP bound we can prove depends on the number of honest Clients: if all clients are honest then we get a great bound; if most are malicious, then we still the remaining Clients still get some degree of protection.

Reply:
That makes sense, if we have a bound that degrades gracefully when the number of honest Clients decreases, then OneAggregator-OneClient is simply a special case of this more general bound. But can't we capture this graceful degradation directly in the OA-MC trust model?

Two things still bug me a little with the current definition of OA, which is really OneAggregator-FewClients.
- For a simple policy like Aggregator-side noise, the bound does not even depend on the number of honest Clients. So aggregator-side noise is secure under OneAggregator-OneClient. It seems wasteful to request stronger (and possibly more vague) trust assumptions in the form of OneAggregator-FewClients, when we really only need OneAggregator-OneClient.
- Currently, the separation between OA and OA-MC seems a bit blurry: when do we go from "most Clients are honest" (OA-MC) to "most Clients are malicious" (OA)? Do we have protocols that drastically change when we go from 51% honest to 49% honest? I'd expect this kind of separation (maybe with a 2/3 threshold instead of 1/2) when we do MPC, but I don't think we're considering these protocols for now.

Reply:
>     * For a simple policy like Aggregator-side noise, the bound does not even depend on the number of honest Clients. So aggregator-side noise is secure under OneAggregator-OneClient. It seems wasteful to request stronger (and possibly more vague) trust assumptions in the form of OneAggregator-FewClients, when we really only need OneAggregator-OneClient.


I agree that this distinction doesn't matter. However the distinction between OA-OC and OA-MC is important because, if you're targeting OA-MC, but it turns out that you're in OA-OC, then you may not have any privacy at all, say, if you're doing Client-only noise. On the other hand, if you're targeting OA-OC and use Aggregator-only noise, then you may end up with more utility loss than had you targeted OA-MC.

I've changed "OA" to "OAOC" as you suggest.

>     * Currently, the separation between OA and OA-MC seems a bit blurry: when do we go from "most Clients are honest" (OA-MC) to "most Clients are malicious" (OA)? Do we have protocols that drastically change when we go from 51% honest to 49% honest? I'd expect this kind of separation (maybe with a 2/3 threshold instead of 1/2) when we do MPC, but I don't think we're considering these protocols for now.

We are considering protocols like DPrio, where the privacy degrades as the number of corrupt clients increases. We're also considering pure Client-noise, which should similarly degrade gracefully. I don't expect we'll have schemes where the DP bound suddenly plummets once one more Client starts cheating!

Reply:
Great, thanks @cjpatton!